src/core/providers/qgsabstractdatabaseproviderconnection.h

```diff
@@ -197,6 +215,21 @@ class CORE_EXPORT QgsAbstractDatabaseProviderConnection : public QgsAbstractProv
 
     };
 
+    /**
+     * \brief The SqlVectorLayerOptions stores all information required to create a SQL (query) layer.
+     * \see createSqlVectorLayer()
+     *
+     * \since QGIS 3.22
+     */
+    struct CORE_EXPORT SqlVectorLayerOptions
+    {
+      QString sql; //!< The SQL expression that defines the SQL (query) layer
+      QString filter; //!< Additional subset string (provider-side filter), not all data providers support this feature: check support with SqlLayerDefinitionCapability::Filters capability
+      QString layerName; //!< Optional name for the new layer
+      QStringList primaryKeyColumns; //!< List of primary key column names
+      QString geometryColumn; //!< Name of the geometry column
+      bool disableSelectAtId; //!< If SelectAtId is disabled (default is false), not all data providers support this feature: check support with SqlLayerDefinitionCapability::SelectAtId capability
```

Comment:
```suggestion
      bool disableSelectAtId = false; //!< If SelectAtId is disabled (default is false), not all data providers support this feature: check support with SqlLayerDefinitionCapability::SelectAtId capability
```

Reply:
I've always found "SelectAtId" to be a horribly named thing. I'm still not 100% sure what this actually means, even after staring at the QGIS code for more than 10 years... :scream: . Could we use a better wording here?

Reply:
To my understanding certain providers allow to set this flag for certain layers (views?) that have no way to select a feature by ID, this makes it possible to optimize certain code paths. 

I have no idea about the wording, maybe the capability flag should be `DisableSelectAtId` (I prefer this over `DisableSelectById` because it avoid confusion with Qgs selectById methods).

Reply:
In that case I'd probably go with something like "UnstableFeatureIds" ?